client/my-sites/marketing/connections/service-action.jsx

```diff
@@ -51,9 +51,7 @@ const SharingServiceAction = ( {
 		} else {
 			label = translate( 'Disconnect', { context: 'Sharing: Publicize disconnect button label' } );
 		}
-		if ( 'must-disconnect' === status ) {
-			warning = true;
-		}
+		warning = true;
```

Comment:
This is interesting. I checked Calypso to see what is currently done, and 'Disconnect' is styled the same as 'Connect', as it is in Cloud.

This warning state is for when a connection is broken, in order to alert someone to the fact they need to reconnect. That is more likely to be missed now if we style the disconnect button the same.

@keoshi Do you have any thoughts on how we should differentiate those two states?

Reply:
How about an extra exclamation mark icon for the broken connection button?

Reply:
If the service was connected before and became disconnected for some reason, that indicates the product is not working as the customer expects to. In other words, something changed that needs their immediate attention.

The user has shown intent when setting it up initially, so I'd consider “Reconnect” a primary primary action when compared to “Connect” (for the first time) or “Disconnect”, like so:

<img width="1056" alt="image" src="https://user-images.githubusercontent.com/390760/179732413-49afc37a-cfb3-4af8-8987-d4963f0f64cd.png">

_However_...because of the different button styles we have in JP cloud it's starting to look like stoplights a little bit, so I wouldn't be opposed to using an icon as @danielpost suggested as a way to create that differentiation.

Reply:
I like the icon idea, if there is one that can be shared between the two environments. I'm not sure that Disconnect and Reconnect should be the same colour, so maybe using the primary style is better? I wonder if Connect should be primary, Disconnect secondary, and Reconnect to be the scary/warning state.

Reply:
> I like the icon idea, if there is one that can be shared between the two environments.

Yeah, we can use the `Gridicon` component inside a `Button` and it should behave as documented here: https://wpcalypso.wordpress.com/devdocs/design/button

As for the icon, I'd lean towards one of these two:

Notice | Sync
--|--
`notice-outline`: an exclamation mark inside a circle, denoting something that needs immediate attention. | `sync`: a circling double-arrow, hinting at the refresh action that should be performed.
<img width="46" alt="image" src="https://user-images.githubusercontent.com/390760/179779892-165de56c-fd33-422a-b90a-344445054d22.png"> | <img width="55" alt="image" src="https://user-images.githubusercontent.com/390760/179780010-337dbe39-70ee-4675-afbe-f883fbf306a5.png">

> I wonder if Connect should be primary, Disconnect secondary, and Reconnect to be the scary/warning state.

As a rule of thumb, there should only be one primary CTA on any given page to make sure there's always a single recommended action.

I also don't see Reconnect as a scary action at all, but one we should encourage users to take, especially when considering it was working before. It should communicate reassurance that we're taking a close look at connections rather than fear.

Reply:
> As a rule of thumb, there should only be one primary CTA

Ah yes, that makes sense. There could be more than one reconnect button too, so that suggests that the icon is the best option.

Reply:
I played around a bit with these ideas:

![CleanShot 2022-07-20 at 10 37 00](https://user-images.githubusercontent.com/36671565/179938345-be955d52-868b-46ae-bb89-72813720ce52.png)

![CleanShot 2022-07-20 at 10 43 41](https://user-images.githubusercontent.com/36671565/179938568-291c802d-0320-43f3-981b-f0f9d69e8fd9.png)

![CleanShot 2022-07-20 at 10 32 41](https://user-images.githubusercontent.com/36671565/179938338-270a71e8-8dc1-499f-a2fb-20f2cf48a322.png)

![CleanShot 2022-07-20 at 10 34 37](https://user-images.githubusercontent.com/36671565/179938343-2ddf35b6-893d-4f18-8666-08c943417097.png)

![CleanShot 2022-07-20 at 10 37 27](https://user-images.githubusercontent.com/36671565/179938350-df238052-81d6-442c-8f09-197f4431d108.png)

What do you think is best? Or maybe combining one with the other? I feel like the green primary buttons look a bit aggressive compared to the minimalistic design of the page, I would scratch those.

Reply:
I think the third option is my favorite, followed by the second one. I agree about the green primary buttons, especially considering most people won't even want to connect to more than one network.

Reply:
Either first or last for me. We should not be using a "destructive" styling for something that is actually the recommended action.

Reply:
Resolved in p1658412564152429-slack-C02JJ910CNL